src/gtc/passes/gtir_upcaster.py

```diff
@@ -58,7 +58,7 @@ def _numpy_ufunc_upcasting_rule(*dtypes, ufunc: np.ufunc):
         inputs, output = t.split("->")
         assert len(inputs) == len(dtypes)
         if all(
-            np.can_cast(data_type_to_typestr(arg_dtype), cand_typestr)
+            arg_dtype <= typestr_to_data_type(np.dtype(cand_typestr).str)
```

Comment:
The docstring might need a small touchup to reflect that we're not quite using `np.can_cast`.

Reply:
Done. And noticed that I also have to fix the function below.